File: apps/mobile/lib/core/config/feature_flags.dart
// Feature flags: keep mobile, web, and database defaults aligned.
// Only the informational catalog is enabled; future integrations stay OFF.
library;

enum FeatureFlag {
  linkedAccounts,
  bankSync,
  ewalletSync,
  transfers,
  transferRecipientValidation,
  pushNotifications,
  institutionCatalog,

  /// Requires the hosted Supabase confirmation template to contain {{ .Token }}.
  emailOtpVerification,
}

class FeatureFlags {
  const FeatureFlags._();

  static const FeatureFlags instance = FeatureFlags._();

  static const Map<FeatureFlag, bool> _defaults = {
    FeatureFlag.linkedAccounts: false,
    FeatureFlag.bankSync: false,
    FeatureFlag.ewalletSync: false,
    FeatureFlag.transfers: false,
    FeatureFlag.transferRecipientValidation: false,
    FeatureFlag.pushNotifications: false,
    FeatureFlag.institutionCatalog: true,
    FeatureFlag.emailOtpVerification: true,
  };

  static bool isEnabled(FeatureFlag flag, {Map<FeatureFlag, bool>? overrides}) {
    if (overrides != null && overrides.containsKey(flag)) {
      return overrides[flag]!;
    }
    return _defaults[flag] ?? false;
  }

  static bool get linkedAccountsEnabled =>
      isEnabled(FeatureFlag.linkedAccounts);
  static bool get transferEnabled => isEnabled(FeatureFlag.transfers);
}

File: packages/config/src/featureFlags.ts
import { FEATURE_FLAGS, type FeatureFlagId } from "@nusarta/types";

// Feature flags: keep web, mobile, and database defaults aligned.
// Future integrations remain disabled until backend and provider support exist.
export const defaultFeatureFlags: Record<FeatureFlagId, boolean> = {
  linked_accounts: FEATURE_FLAGS.linked_accounts.defaultEnabled,
  bank_sync: FEATURE_FLAGS.bank_sync.defaultEnabled,
  ewallet_sync: FEATURE_FLAGS.ewallet_sync.defaultEnabled,
  transfers: FEATURE_FLAGS.transfers.defaultEnabled,
  transfer_recipient_validation:
    FEATURE_FLAGS.transfer_recipient_validation.defaultEnabled,
  push_notifications: FEATURE_FLAGS.push_notifications.defaultEnabled,
  institution_catalog: FEATURE_FLAGS.institution_catalog.defaultEnabled,
};

/** Lightweight in-app flag lookup with a provided (backend) map. */
export function isFeatureEnabled(
  feature: FeatureFlagId,
  overrides: Partial<Record<FeatureFlagId, boolean>> = {}
): boolean {
  return overrides[feature] ?? defaultFeatureFlags[feature];
}
